spec/models/course/condition/achievement_spec.rb

```diff
@@ -20,12 +22,26 @@
 
       context "when an achievement is already included in its conditional's conditions" do
         subject do
-          create(:course_condition_achievement, :duplicate_child, course: course).tap do |cca|
-            allow(cca).to receive(:achievement_id_changed?).and_return(true)
+          create(:course_condition_achievement, course: course).tap do |achievement_condition|
+            allow(achievement_condition).to receive(:achievement_id_changed?).and_return(true)
           end
         end
         it { is_expected.to_not be_valid }
       end
+
+      context 'when an achievement is required by another conditional with the same id' do
+        subject do
+          id = Time.now.to_i
```

Comment:
Yeah, but timestamp is a big integer so less chance to crash. Any suggestion how to find an available id in test?